src/main/java/com/jprotractor/NgWebDriver.java

```diff
@@ -11,137 +10,125 @@
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.internal.WrapsDriver;
 
-public class NgWebDriver implements WebDriver, WrapsDriver {
-    private WebDriver driver;
-    private JavascriptExecutor jsExecutor;
-    private String rootElement;
-    @SuppressWarnings("unused")
-    private NgModule[] mockModules;
-    public boolean ignoresync;
-    // little max iteration count for protector
-    public int maxIterations;
-    private int iterationCount;
-
-    public NgWebDriver(WebDriver driver) {
-        this.driver = driver;
-        this.jsExecutor = (JavascriptExecutor) driver;
-        this.rootElement = "body";
-        maxIterations = 30;
-        iterationCount = 0;
-    }
-
-    public NgWebDriver(final WebDriver driver, final boolean ignoreSync) {
-        this(driver);
-        this.ignoresync = ignoreSync;
-    }
-
-    public NgWebDriver(final WebDriver driver, final NgModule[] mockModules) {
-        this(driver, "body", mockModules);
-    }
-
+/**
+ * Angular WebDriver Implementation.
+ */
+public final class NgWebDriver implements WebDriver, WrapsDriver {
+    private final WebDriver driver;
+    private final JavascriptExecutor jsExecutor;
+    private final String root;
+
+    /**
+     * Ctor.
+     * @param drv Parent web driver.
+     */
+    public NgWebDriver(final WebDriver drv) {
+        this(drv, "body");
+    }
+
+    /**
+     * Ctor.
+     * @param drv Parend Web Driver.
+     * @param rootel Root Element.
+     */
     public NgWebDriver(
-        final WebDriver driver,
-        final String rootElement,
-        final NgModule[] mockModules
+        final WebDriver drv,
+        final String rootel
     ) {
-        if (!(driver instanceof JavascriptExecutor)) {
+        if (!(drv instanceof JavascriptExecutor)) {
             throw new WebDriverException(
                 "The WebDriver instance must implement the " +
                     "JavaScriptExecutor interface."
             );
         }
 
-        this.driver = driver;
-        this.jsExecutor = (JavascriptExecutor) driver;
-        this.rootElement = rootElement;
-        this.mockModules = mockModules;
+        this.driver = drv;
+        this.jsExecutor = (JavascriptExecutor) drv;
+        this.root = rootel;
     }
 
+    @Override
     public WebDriver getWrappedDriver() {
         return this.driver;
     }
 
+    @Override
     public void close() {
         this.driver.close();
-
     }
 
-    public NgWebElement findElement(By by) {
+    @Override
+    public NgWebElement findElement(final By by) {
         this.waitForAngular();
         return new NgWebElement(this, this.driver.findElement(by));
     }
 
-    public List<NgWebElement> findNGElements(By by) {
-        this.waitForAngular();
```

Comment:
I do not have tests for this method yet (cascading drop downs with $watch may be such a test), i think it may be worth keeping  due to explicit waitForAngular()